modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -568,8 +568,7 @@ public void processOpts() {
         this.sanitizeConfig();
 
         // optional jackson mappings for BigDecimal support
-        importMapping.put("ToStringSerializer", "com.fasterxml.jackson.databind.ser.std.ToStringSerializer");
-        importMapping.put("JsonSerialize", "com.fasterxml.jackson.databind.annotation.JsonSerialize");
+        importMapping.put("JsonFormat", "com.fasterxml.jackson.annotation.JsonFormat");
```

Comment:
how will this change impact the auto-generated code if bigDecimalAsString is not enabled (set to false by default)?

shall we add a if-then-else to check if bigDecimalAsString is enabled and update the importMapping accordingly?

Reply:
It's a good point. I thought `importMapping` was conditional to the import being added by `model.imports.add("JsonFormat");` (https://github.com/OpenAPITools/openapi-generator/pull/19322/files#diff-a98cba8ecadaf5a662edc508aa5857972339552ec6bd6e4216680c7c6070e643R1710)

I can also see that there some ad-hoc places where it manually removes this import from other mappings, like https://github.com/OpenAPITools/openapi-generator/blob/d1b9f9284a007da3a6f92fcad5fc966709be5454/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaCXFServerCodegen.java#L230-L231), so it seems a good change to add an if at this point.

I am not convinced that `JsonSerialize` is used only for BigDecimal support as the comment seems to indicate. Will have a look at this as well.
Thanks for your comment @wing328

Reply:
@wing328 I have pushed some commits that I believe resolve your concerns, let me know your thoughts